src/server/pmix_server_get.c

```diff
@@ -429,20 +459,18 @@ pmix_status_t pmix_server_get(pmix_buffer_t *buf,
      * whomever is hosting the target process */
     if (NULL != pmix_host_server.direct_modex) {
         rc = pmix_host_server.direct_modex(&lcd->proc, info, ninfo, dmdx_cbfunc, lcd);
-        /* if they specified a timeout, set it up now */
-        if (0 < tv.tv_sec) {
-            pmix_event_evtimer_set(pmix_globals.evbase, &req->ev,
-                                   get_timeout, req);
-            pmix_event_evtimer_add(&req->ev, &tv);
-            req->event_active = true;
+        if (PMIX_SUCCESS != rc) {
+            /* may have a function entry but not support the request */
+            PMIX_INFO_FREE(info, ninfo);
+            pmix_list_remove_item(&pmix_server_globals.local_reqs, &lcd->super);
```

Comment:
I am not sure this can be released here(at line 835 will trigger warning this is not in the list),  line 836 will cause segment fault

Reply:
Also, from what i tested,  the return value rc at line:461 is IN_PROGRESS, then this returns to pmix_server.c at line:3332 and cd will be released. Since the request is not satisfied, can we release cd ?

Reply:
I don't understand what you are describing - PRRTE master clearly returns PMIX_SUCCESS, not IN_PROGRESS, as that is what the standard requires. What are you looking at?

Reply:
> I am not sure this can be released here(at line 835 will trigger warning this is not in the list), line 836 will cause segment fault

Again, I don't understand what you are describing. We remove it from the list and release it, then return. Line 835 gets the number of elements on the list, and line 836 goes to the "cleanup" label - so how can that cause a segfault??

Reply:
Sorry i was in review mode, the line number is not correct,  when i did run the code,  at line 866 867 will cause segfault.

Reply:
I did rebased my code on master, but i realized i should do that on V3.0, I will test after i fix my repo.  Sorry about my mistake.

Reply:
No problem - let me know if you still see a problem once you get sync'd.